foam_controlwrapper/foam_mesh.py

```diff
@@ -0,0 +1,772 @@
+                + str(cell.uid)
+            raise KeyError(error_str)
+
+        if not isinstance(cell, Cell):
+            error_str = "Trying to update an object with the wrong type. "\
+                + "Cell expected."
+            raise TypeError(error_str)
+
+# this not included for performance reasons
+#        pointLabels = []
+#        for point in cell.points:
+#            pointLabels.append(self._uuidToFoamLabel[point.uid])
+#        foamface.setCellPoints(self.name,
+#                               self._uuidToFoamLabel[cell.uid],
+#                               pointLabels)
+
+# test if data exists and if not create (must be changed in the future,
+# while not efficient to make this in cell level)
+        foamFiles = FoamFiles()
+        dataNames = foamFiles.get_cell_data_names(self.path,
+                                                  self.time)
+
+        for data in cell.data:
+            if data == CUBA.PRESSURE:
+                dataName = "p"
+                if dataName not in dataNames:
+                    self.create_dummy_celldata(dataName,
+                                               [0, 2, -2, 0, 0, 0, 0])
+                foamFiles.set_cell_data(self.path, self.time,
+                                        dataName,
+                                        self._uuidToFoamLabel[cell.uid],
+                                        cell.data[data], 'scalar')
+                dataName = "p_rgh"
+                if dataName not in dataNames:
+                    self.create_dummy_celldata(dataName,
+                                               [0, 2, -2, 0, 0, 0, 0])
+                foamFiles.set_cell_data(self.path, self.time,
+                                        dataName,
+                                        self._uuidToFoamLabel[cell.uid],
+                                        cell.data[data], 'scalar')
+            elif data == CUBA.VOLUME_FRACTION:
+                dataName = "alpha1"
+                if dataName not in dataNames:
+                    self.create_dummy_celldata(dataName,
+                                               [0, 0, 0, 0, 0, 0, 0])
+                foamFiles.set_cell_data(self.path, self.time,
+                                        dataName,
+                                        self._uuidToFoamLabel[cell.uid],
+                                        cell.data[data], 'scalar')
+            elif data == CUBA.VELOCITY:
+                dataName = "U"
+                if dataName not in dataNames:
+                    self.create_dummy_cellvectordata(dataName,
+                                                     [0, 1, -1, 0, 0, 0, 0])
+                foamFiles.set_cell_data(self.path, self.time,
+                                        dataName,
+                                        self._uuidToFoamLabel[cell.uid],
+                                        tuple(cell.data[data]), 'vector')
+            elif data == "":
```

Comment:
In this context `data` is a CUBA key, so it should be not possible to have a value of "". It looks that this check is unnecessary.